src/output_writers.jl

```diff
@@ -392,19 +392,29 @@ VerticalPlanes(model) = VerticalPlanes(model.arch, model.grid)
 mutable struct Checkpointer <: OutputWriter
                  dir :: String
               prefix :: String
-    output_frequency :: Int
+           frequency :: Int
+            interval :: FT
+             structs :: S
+           fieldsets :: F
 end
 
-function Checkpointer(; output_frequency, dir=".", prefix="checkpoint", force=false)
+function Checkpointer(; output_frequency, dir=".", prefix="checkpoint",
+                      checkpointed_structs = (:arch, :boundary_conditions, :grid, :clock, :eos, :constants, :closure),
+                      checkpointed_fieldsets = (:velocities, :tracers, :G, :Gp),
+                      force=false)
     mkpath(dir)
     return Checkpointer(dir, prefix, output_frequency)
 end
 
 function savesubfields!(file, model, name, flds=propertynames(getproperty(model, name)))
-    for f in flds
-        file["$name/$f"] = Array(getproperty(getproperty(model, name), f).data.parent)
+    if name ∉ (:forcing)
```

Comment:
I'm a bit confused about how this function works:

- why restrict this functionality to `Field` (as it looks to me), but provide a hook when `name=:forcing`?
- can't we check the `type` of the object being saved? Other fields (like boundary conditions, or future fields that we add to model) may be functions.

I think it'd be better to write a function that dispatches on the type of the object being saved, eg

```julia
savefield(file, location, field) = file[location] = field
savefield(file, location, field::AbstractArray) = file[location] = Array(field)
savefield(file, location, field::Field) = file[location] = Array(field.data.parent)
savefield(file, location, field::Function) = warn("Cannot checkpoint Function object into $location!")
```

This seems more robust and general to me.